common/src/main/kotlin/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketCoreImpl.kt

```diff
@@ -520,14 +551,22 @@ internal class WebSocketCoreImpl(
         }
         // If we still have flows flowing
         val hasFlowMessages = synchronized(activeMessages) { activeMessages.any { it.value.eventFlow != null } }
-        if (hasFlowMessages && ioScope.isActive) {
-            ioScope.launch {
+        if (hasFlowMessages && wsScope.isActive) {
+            wsScope.launch {
+                // TODO Why a delay ? and why 10s I guess you want to wait for the logic above to finish but 10s is quite arbitrary
+                // You could wait for the job above to finish instead
```

Comment:
@jpelgrom I just found out that when you turn your phone into airplane mode or if you loose the connection we retry every 10s is that the expected behavior? And 10s is to avoid trying too often?

Reply:
It's a nice feature but it means every 10s we try to reconnect. I do wonder if it impacts the battery usage.

Reply:
The delay is there for short interruptions of the connection, for example if you switch access points or when switching from Wi-Fi to mobile data. Very basic but it worked well enough. There is a battery impact but shouldn't be significant beause retrying only happens if there were subscriptions, so the app was already keeping a connection alive.

Reply:
Looking at making this 'smarter' in a future PR is fine by me, your question was just why is it 10s :)